staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -65,6 +68,7 @@ func NewRecommendedOptions(prefix string, codec runtime.Codec, processInfo *Proc
 		Audit:                      NewAuditOptions(),
 		Features:                   NewFeatureOptions(),
 		CoreAPI:                    NewCoreAPIOptions(),
+		FeatureGate:                featuregate.NewFeatureGate(), // empty default features
```

Comment:
Trying to think what the implications of this are... having multiple feature gate instances floating around seems bad. Are we overriding this with our existing util featureGate buckets anywhere?

Reply:
> Trying to think what the implications of this are... having multiple feature gate instances floating around seems bad. Are we overriding this with our existing util featureGate buckets anywhere?

kube-apiserver does.  I could choose to reference the on in the apiserver package.

Reply:
fixing the feature gate impl to error if an unregistered feature is interrogated seems like a good path forward. https://github.com/kubernetes/kubernetes/pull/84865

Reply:
it is a little strange that the default here will not have the generic apiserver features registered

Reply:
> it is a little strange that the default here will not have the generic apiserver features registered

to my knowledge, our recommend options don't include featuregates

Reply:
https://github.com/kubernetes/kubernetes/blob/62f66ea2f5e76dfe172f242a7d2f859b25338282/staging/src/k8s.io/apiserver/pkg/features/kube_features.go#L157-L167

those?

Reply:
We have them defined, but we don't actually expose a way to wire them to options in the genericapiserver.  Or at least I don't remember them in recommendedoptions

Reply:
they're not wired to options, but the API server still checks them in handlers, etc, so if they're not registered, I'm pretty sure the API server wouldn't run